packages/components/src/templates/next/components/complex/KeyStatistics/KeyStatistics.tsx

```diff
@@ -78,6 +87,19 @@ const KeyStatistics = ({ title, statistics, layout }: KeyStatisticsProps) => {
           </div>
         ))}
       </div>
+
+      {url !== undefined && url !== "" && (
+        <div className={compoundStyles.urlButtonContainer()}>
+          <LinkButton
+            href={getReferenceLinkHref(url, site.siteMap)}
```

Comment:
**note:** i didn't look into `getReferenceLinkHref` but the name here sounds like it shuold only allow internal links, which have a format of `[ref,$siteId,$resourceId]`. this is in direct contrast to the link format, which also allows normal urls + `mailto`

Reply:
Yeah I haven't looked very deep into this too but copied this from infobar component

Reply:
issue: i went to look at this function definition and this function doesn't quite work here **unless we strictly allow internal links**. this is because this function does a regex for `[resource, <number>, < number>]` then does a lookup for this number to generate the permalink.

Reply:
thanks! will fix together once the other PR is merged and we can pass in linktypes